src/Aggregation/Reduced.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ElegantBro\Arrayee\Aggregation;
+
+use ElegantBro\Arrayee\Just;
+use ElegantBro\Interfaces\Arrayee;
+use Exception;
+use function array_reduce;
+
+final class Reduced implements Arrayee
+{
+    /**
+     * @var Arrayee
+     */
+    private $arrayee;
+
+    /**
+     * @var callback
+     */
+    private $callback;
+
+    /**
+     * @var Arrayee
+     */
+    private $initial;
+
+    /**
+     * Reduced constructor.
+     * @param Arrayee $arrayee
+     * @param callable(array, mixed): array $callback function(array $carry, $item): array
+     * @param Arrayee $initial
+     */
+    public function __construct(Arrayee $arrayee, callable $callback, Arrayee $initial)
+    {
+        $this->arrayee = $arrayee;
+        $this->callback = $callback;
+        $this->initial = $initial;
+    }
+
+    /**
+     * @param Arrayee $arrayee
+     * @param callable(array, mixed): array $callback function(array $carry, $item): array
```

Comment:
the same about type, and by convenience secondary constructors placed above the primary

Reply:
@stepanets I've fixed it both.